packages/fleather/lib/src/widgets/editor_input_client_mixin.dart

```diff
@@ -120,51 +114,38 @@ mixin RawEditorStateTextInputClientMixin on EditorState
   AutofillScope? get currentAutofillScope => null;
 
   @override
-  void updateEditingValue(TextEditingValue value) {
+  void updateEditingValueWithDeltas(List<TextEditingDelta> textEditingDeltas) {
     if (!shouldCreateInputConnection) {
       return;
     }
 
-    if (_lastKnownRemoteTextEditingValue == value) {
-      // There is no difference between this value and the last known value.
-      return;
-    }
-
-    // Check if only composing range changed.
-    if (_lastKnownRemoteTextEditingValue!.text == value.text &&
-        _lastKnownRemoteTextEditingValue!.selection == value.selection) {
-      // This update only modifies composing range. Since we don't keep track
-      // of composing range in Zefyr we just need to update last known value
-      // here.
-      // This check fixes an issue on Android when it sends
-      // composing updates separately from regular changes for text and
-      // selection.
-      _lastKnownRemoteTextEditingValue = value;
-      return;
+    for (final textEditingDelta in textEditingDeltas) {
+      _lastKnownRemoteTextEditingValue =
+          textEditingDelta.apply(_lastKnownRemoteTextEditingValue!);
+      if (textEditingDelta is TextEditingDeltaNonTextUpdate) {
+        widget.controller.updateSelection(textEditingDelta.selection);
+      } else {
+        int start = 0, length = 0;
+        String data = '';
+        if (textEditingDelta is TextEditingDeltaInsertion) {
+          start = textEditingDelta.insertionOffset;
+          data = textEditingDelta.textInserted;
+        } else if (textEditingDelta is TextEditingDeltaDeletion) {
+          start = textEditingDelta.deletedRange.start;
+          length = textEditingDelta.deletedRange.length;
+        } else if (textEditingDelta is TextEditingDeltaReplacement) {
+          start = textEditingDelta.replacedRange.start;
+          length = textEditingDelta.replacedRange.length;
+          data = textEditingDelta.replacementText;
+        }
+        widget.controller.replaceText(start, length, data);
```

Comment:
I think you need to add the new selection coming form the delta when updating the controller
```dart
widget.controller.replaceText(start, length, data, selection: textEditingDelta.selection);
```
This seems to work but might need some more tests

There seems to be a bug with the `suppr` case, the `deletedRange` is the same as a `del` when it should be `TextRange(start:deletedRange.start + 1, end: deletedRange.end + 1)`

Reply:
I couldn't understand what do you mean by `suppr` case? Also I didn't face any issue with deletion. @amantoux

Reply:
When you hit `suppr` instead of `del` on keyboard, the behavior is incorrect in my tests.
But you have to run it on web (just tried on macOS and there is no issue there)